src/components/reservations/BikeTable.js

```diff
@@ -0,0 +1,239 @@
+      selectedBikeIds = selectedBikeIds.concat(selected.slice(0, selectedIndex), selected.slice(selectedIndex + 1));
+    }
+
+    setSelected(selectedBikeIds);
+    onBikeSelection(selectedBikeIds);
+  };
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const isSelected = (name) => selected.indexOf(name) !== -1;
+
+  const emptyRows = rowsPerPage - Math.min(rowsPerPage, rows.length - page * rowsPerPage);
+
+  BikeTable.propTypes = {
+    bikes: PropTypes.any,
+    onBikeSelection: PropTypes.func,
+  };
+
+  return (
+    <div className={classes.root}>
+      <Paper className={classes.paper}>
+        <TableContainer>
+          <Table className={classes.table} aria-labelledby="tableTitle" aria-label="enhanced table">
+            <EnhancedTableHead classes={classes} order={order} orderBy={orderBy} onRequestSort={handleRequestSort} />
+            <TableBody>
+              {stableSort(rows, getComparator(order, orderBy))
+                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                .map((row, index) => {
+                  const isItemSelected = isSelected(row.id);
+                  const labelId = `enhanced-table-checkbox-${index}`;
+
+                  return (
+                    <TableRow
+                      hover
+                      onClick={(event) => handleClick(event, row.id)}
+                      role="checkbox"
+                      aria-checked={isItemSelected}
+                      tabIndex={-1}
+                      key={row.id}
+                      selected={isItemSelected}
+                    >
+                      <TableCell padding="checkbox">
+                        <Checkbox checked={isItemSelected} inputProps={{ 'aria-labelledby': labelId }} />
+                      </TableCell>
+                      <TableCell align="right" component="th" id={labelId} scope="row" padding="none">
+                        {row.type}
+                      </TableCell>
+                      <TableCell align="right">{row.isElectric}</TableCell>
+                      <TableCell align="right">{row.size}</TableCell>
+                      <TableCell align="right">{row.location}</TableCell>
+                    </TableRow>
+                  );
+                })}
+              {emptyRows > 0 && (
+                <TableRow>
+                  <TableCell colSpan={6} />
+                </TableRow>
+              )}
+            </TableBody>
+          </Table>
+        </TableContainer>
+        <TablePagination
+          rowsPerPageOptions={[5, 10, 25]}
+          component="div"
+          count={rows.length}
+          rowsPerPage={rowsPerPage}
+          page={page}
+          onPageChange={handleChangePage}
+          onRowsPerPageChange={handleChangeRowsPerPage}
+        />
+      </Paper>
+    </div>
+  );
+};
```

Comment:
239 lines for the component file it must be a mistake ;) but if it worked in this form let Mikołaj do refactor of this and then you will get an example he to work with such things in the future. ok?

Reply:
ok